src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java

```diff
@@ -427,7 +427,7 @@ protected void retrieve(@CheckForNull SCMSourceCriteria criteria,
                         listener.getLogger().println("Checking branches...");
                         int count = 0;
                         for (final SCMRevision revision : revisions) {
-                            if (!(revision instanceof SCMRevisionImpl)) {
+                            if (!(revision instanceof SCMRevisionImpl) || (revision instanceof GitTagSCMRevision)) {
```

Comment:
@MarkEWaite will probably move to a dedicated BranchSCMHead for Git plugin 4.0.0 (and use the SCMHeadMigration functionality to do it) as this bug wouldn't have happened